openmc/model/triso.py

```diff
@@ -91,6 +92,363 @@ def classify(self, lattice):
+
+    def nearby_mesh_cells(self, p):
+        """Calculates the indices of all cells in a mesh overlaid on the domain
+        within one diameter of the given particle.
+
+        Parameters
+        ----------
+        p : Iterable of float
+            Cartesian coordinates of particle center.
+
+        Returns
+        -------
+        list of tuple of int
+            Indices of mesh cells.
+
+        """
+        d = 2*self.particle_radius
+        r = [[a/self.cell_length[i] for a in [p[i]-d, p[i], p[i]+d]]
+             for i in range(3)]
+        return list(itertools.product(*({int(x) for x in y} for y in r)))
+
+    @abstractmethod
+    def reset(self):
+        """Recalculate attributes that depend on input parameters if any of the
+        parameters are modified.
+
+        """
+        pass
+
+    @abstractmethod
+    def random_point(self):
+        """Generate Cartesian coordinates of center of a particle that is
+        contained entirely within the domain with uniform probability.
+
+        Returns
+        -------
+        list of float
+            Cartesian coordinates of particle center.
+
+        """
+        pass
+
+
+class _CubicDomain(_Domain):
+    """Cubic container in which to pack particles.
+
+    Parameters
+    ----------
+    length : float
+        Length of each side of the cubic container.
+    particle_radius : float
+        Radius of particles to be packed in container.
+    center : Iterable of float
+        Cartesian coordinates of the center of the container. Default is
+        [0., 0., 0.]
+
+    Attributes
+    ----------
+    length : float
+        Length of each side of the cubic container.
+    particle_radius : float
+        Radius of particles to be packed in container.
+    center : list of float
+        Cartesian coordinates of the center of the container. Default is
+        [0., 0., 0.]
+    cell_length : list of float
+        Length in x-, y-, and z- directions of each cell in mesh overlaid on
+        domain.
+    limits : list of float
+        Minimum and maximum position in x-, y-, and z-directions where particle
+        center can be placed.
+    volume : float
+        Volume of the container.
+
+    """
+
+    def __init__(self, length, particle_radius, center=[0., 0., 0.]):
+        self._length = None
+        super(_CubicDomain, self).__init__(particle_radius, center)
+        self.length = length
+
+    @property
+    def volume(self):
+        return self.length**3
+
+    @property
+    def length(self):
+        return self._length
+
+    @length.setter
+    def length(self, length):
+        self._length = float(length)
+        self.reset()
+
+    def reset(self):
+        if (self.particle_radius is not None and self.center is not None
+            and self.length is not None):
+            xlim = self.length/2 - self.particle_radius
+            self.limits = [[x - xlim for x in self.center],
+                           [x + xlim for x in self.center]]
```

Comment:
I would make `limits` and `cell_length` computed properties, i.e. have something like:
```Python
@property
def cell_length(self):
    if self._cell_length is None:
        self._cell_length = [x/int(x/(4*self.particle_radius))
                             for x in mesh_length]
    return self._cell_length
```
Then just change the setters for `length` and `particle_radius` to reset the internal `_cell_length` and `_limits` variables to `None` so that the cell length and limits would be re-computed whenever the length or particle radius changes. That would make it so that you don't need the `reset()` method at all. Does that make sense?